### PR TITLE
Add explain API to REST API

### PR DESCRIPTION
**Pull request by ashwinkumar12345**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/rest-api/explain.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+# Explain
+Introduced 1.0
+{: .label .label-purple }
+
+Wondering why a specific document ranks higher (or lower) for a query? You can use the explain API for an explanation of how the relevance score (`_score`) is calculated for every result.
+
+OpenSearch uses a probabilistic ranking framework called [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) to calculate relevance scores. Okapi BM25 is based on the original [TF/IDF](http://lucene.apache.org/core/4_0_0/core/org/apache/lucene/search/package-summary.html#scoring) framework used by Apache Lucene.
+
+The explain API is an expensive operation in terms of both resources and time. We recommend you don't use it for production workloads.
```
Comment (aetter) [inline _opensearch/rest-api/explain.md:16]: I think we can reword this second sentence a bit -- surely it's appropriate to use it for one-off requests. Maybe something like "On production clusters, we recommend using it sparingly for the purpose of troubleshooting."?

File: _opensearch/rest-api/explain.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+{
+  "query": {
+    "match": {
+      "customer_first_name": "Mary"
+    }
+  }
+}
+```
+
+To see the explain output for a specific document, specify the document ID in the URL:
```
Comment (aetter) [inline _opensearch/rest-api/explain.md:35]: This seems like the more common use case. Something like, "More often, you want the output for a single document. In that case, specify the document ID in the URL::

**Review by aetter — APPROVED** [review]
Looks great. Some very minor thoughts.

File: _opensearch/rest-api/explain.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Parameter | Type | Description | Required
+:--- | :--- | :--- | :---
+`<index>` | String | Name of the index. You can only specify a single index. | Yes
+`<_id>` | String | A unique identifier to attach to the document. | Yes
+`analyzer` | String | The analyzer to use in the query string. | No
+`analyze_wildcard` | Boolean | Specifies whether to analyze wildcard and prefix queries. Default is false. | No
+`default_operator` | String | Indicates whether the default operator for a string query should be AND or OR. Default is OR. | No
+`df` | String | The default field in case a field prefix is not provided in the query string. | No
+`lenient` | Boolean | Specifies whether OpenSearch should ignore format-based query failures (for example, querying a text field for an integer). Default is false. | No
+`preference` | String | Specifies a preference of which shard to retrieve results from. Available options are `_local`, which tells the operation to retrieve results from a locally allocated shard replica, and a custom string value assigned to a specific shard replica. By default, OpenSearch executes get document operations on random shards. | No
```
Comment (keithhc2) [inline _opensearch/rest-api/explain.md:68]: I think we want "By default, OpenSearch executes the explain operation on random shards." here

**Review by keithhc2 — COMMENTED** [review]
Looks great, just a minor comment

**Review by keithhc2 — APPROVED** [review]

**Pull request merged**